docs/tutorials/text_sentiment_ngrams.ipynb

```diff
@@ -0,0 +1,496 @@
+    "\n",
+    "    def generate_offsets(data_batch):\n",
+    "        offsets = [0]\n",
+    "        for entry in data_batch:\n",
+    "            offsets.append(offsets[-1] + len(entry))\n",
+    "        offsets = torch.tensor(offsets[:-1])\n",
+    "        return offsets\n",
+    "\n",
+    "    label = torch.tensor([entry[0] for entry in batch])\n",
+    "    text = [entry[1] for entry in batch]\n",
+    "    offsets = generate_offsets(text)\n",
+    "    text = torch.cat(text)\n",
+    "    return text, offsets, label"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define a function to train the model and evaluate results."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "torch.utils.data.DataLoader is recommended for PyTorch domain libraries. We use DataLoader here to load AG_NEWS datasets and send it to the model for training/validation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "lines_to_next_cell": 1
+   },
+   "outputs": [],
+   "source": [
+    "from torch.utils.data import DataLoader\n",
+    "\n",
+    "def train_evaluate_func(model, dataset, loss_func, device, batch_size=64, status='train', optimizer=None):\n",
+    "\n",
+    "    _loss = 0\n",
+    "    _acc = 0\n",
+    "\n",
+    "    if status == 'train':\n",
+    "        model.train()\n",
+    "    else:\n",
+    "        model.eval()\n",
+    "\n",
+    "    data = DataLoader(dataset, batch_size=batch_size, shuffle=True,\n",
+    "                      collate_fn=generate_batch, num_workers=1)\n",
+    "\n",
+    "    for i, (text, offsets, label) in enumerate(data):\n",
+    "        if status == 'train':\n",
+    "            optimizer.zero_grad()\n",
+    "\n",
+    "        text, offsets, label = text.to(device), offsets.to(device), label.to(device)\n",
+    "        output = model(text, offsets)\n",
+    "        loss = loss_func(output, label)\n",
+    "        _loss += loss.item()\n",
+    "\n",
+    "        if status == 'train':\n",
+    "            loss.backward()\n",
+    "            optimizer.step()\n",
+    "\n",
+    "        _acc += (output.argmax(1) == label).sum().item()\n",
+    "\n",
+    "    return _loss / len(dataset), _acc / len(dataset)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Split a dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since the original AG_NEWS has no valid dataset, we have to split the training dataset into train/valid sets with the split ratios of 0.7(train) and 0.3(valid)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import random\n",
+    "def split_dataset(dataset, split_ratio=0.7):\n",
+    "    data = [dataset[i] for i in range(len(dataset))]\n",
+    "    random.shuffle(data)\n",
```

Comment:
We have [random_spit](https://pytorch.org/docs/master/_modules/torch/utils/data/dataset.html#random_split) in data.utils for this - This still works, but I think it's more consistent since we're using torch.utils.data.Dataset after all.